src/cosp.F90

```diff
@@ -767,65 +1216,142 @@ function COSP_SIMULATOR(cospIN,cospgridIN,cospOUT,start_idx,stop_idx,debug)
 
     ! ISCCP (icarus) subcolumn simulator
     if (Lisccp_subcolumn .or. Lmodis_subcolumn) then
-       ! Allocate space for local variables
-       allocate(isccpLEVMATCH(Npoints,isccpIN%Ncolumns),                                 &
-                isccp_boxttop(Npoints,isccpIN%Ncolumns),                                 &
-                isccp_boxptop(Npoints,isccpIN%Ncolumns),                                 &
-                isccp_boxtau(Npoints,isccpIN%Ncolumns), isccp_meantbclr(Npoints))
-       ! Call simulator
-       call icarus_subcolumn(isccpIN%npoints,isccpIN%ncolumns,isccpIN%nlevels,           &
-                             isccpIN%sunlit,isccpIN%dtau,isccpIN%dem,isccpIN%skt,        &
-                             isccpIN%emsfc_lw,isccpIN%qv,isccpIN%at,isccpIN%pfull,       &
-                             isccpIN%phalf,isccpIN%frac_out,isccpLEVMATCH,               &
-                             isccp_boxtau(:,:),isccp_boxptop(:,:),                       &
-                             isccp_boxttop(:,:),isccp_meantbclr(:))
-       ! Store output (if requested)
-       if (associated(cospOUT%isccp_boxtau)) then
-          cospOUT%isccp_boxtau(ij:ik,:)  = isccp_boxtau
-       endif
-       if (associated(cospOUT%isccp_boxptop)) then
-          cospOUT%isccp_boxptop(ij:ik,:) = isccp_boxptop
-       endif
-       if (associated(cospOUT%isccp_meantbclr)) then
-          cospOUT%isccp_meantbclr(ij:ik) = isccp_meantbclr
+       if (verbose) print*,'ISCCP Subcol'
+       if (cospIN % cospswathsIN(1) % N_inst_swaths .gt. 0) then
+          if (isccpIN%Npoints .gt. 0) then
+             allocate(isccpLEVMATCH(isccpIN%Npoints,isccpIN%Ncolumns),  &
```

Comment:
The local variable allocation and subsequent simulator call are the same for both normal processing and swathing, whereas storing the output changes based on swathing. Can we refactor so that the impact of swathing is only evident when storing the output? And have the same allocate and simulator calls for both "modes"?